shell/common/tests/shell_test_surface_metal.h

```diff
@@ -0,0 +1,36 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
```

Comment:
Could we just have a single `shell_test_surface_metal.h` (no `shell_test_surface_metal_impl.h`), and just add `shell_test_surface_metal.cc` or `shell_test_surface_metal_impl.mm` to build depending on the GN settings?

Reply:
We want the tests to be able to compile on both Darwin and non-Darwin platforms. Only that on non-Darwin platforms, the test will be skipped via GTEST_SKIP if Metal is not available and attempting to create a Metal surface will fail. shell_test_surface_metal.cc contains the no-op implementation that is used on non-Darwin platforms and shell_test_surface_metal_impl.mm contains the actual implementation on Darwin. Having just shell_test_surface_metal_impl.mm will cause linker errors on Linux and we cannot put ObjC code in a C++ in shell_test_surface_metal.cc.

Reply:
Discussed offline. The difference of those two approaches seems to be: `shell_test_surface_metal_impl.h` makes it easier to add metal only methods (such methods may be able to be just defined in the `.mm` file), while removing `shell_test_surface_metal_impl.h` makes it easier to add methods to `shell_test_surface.h`.